config/crd/snapshot.storage.k8s.io_volumesnapshots.yaml

```diff
@@ -4,21 +4,49 @@ apiVersion: apiextensions.k8s.io/v1beta1
 kind: CustomResourceDefinition
 metadata:
   annotations:
-    controller-gen.kubebuilder.io/version: (devel)
-    api-approved.kubernetes.io: "https://github.com/kubernetes-csi/external-snapshotter/pull/139"
+    controller-gen.kubebuilder.io/version: v0.2.4
   creationTimestamp: null
   name: volumesnapshots.snapshot.storage.k8s.io
 spec:
+  additionalPrinterColumns:
+  - JSONPath: .status.readyToUse
+    description: Indicates if a snapshot is ready to be used to restore a volume.
+    name: ReadyToUse
+    type: boolean
+  - JSONPath: .spec.source.persistentVolumeClaimName
+    description: The PVC from where a dynamically taken snapshot will be created.
+    name: PVCClaim
+    type: string
+  - JSONPath: .spec.source.volumeSnapshotContentName
+    description: The VolumeSnapshotContent from where a snapshot will be created.
+    name: SnapshotClaim
```

Comment:
PVCClaim / SnapshotClaim sounds strange. What about SourcePVC and SourceSnapshotContent?

In addition, do we need `volumeSnapshotContentName` in `kubectl get`? IMO, the number of pre-provisioned snapshots is going to be much smaller than number of dynamically provisioned ones. We can leave PVCClaim (SourcePVC) empty, indicating that this one is special, user will see `status.boundVolumeSnapshotContentName` anyway and if not, they can do `kubectl describe` to get the details.

Reply:
SourcePVC and SourceSnapshotContent sounds good.

Pre-provisioned snapshots is often used in backup and restore use cases so it is important to show volumeSnapshotContentName in the source, which has a different meaning from status.boundVolumeSnapshotContentName.

If there's a way to dynamically display only one of SourcePVC and SourceSnapshotContent depending on which one is not nil, that will be great.  I don't know if it is possible to do that using printerColumns.

Reply:
It's probably more important to show the status content name since we expect them to be the same in valid cases

Reply:
The source field will never be nil.  Can we have a "VolumeSnapshotSource" field which is either PVC or content name and have a "SourceType" field that indicates whether it is PVC name or content name?

Reply:
I've updated the column names to be `SourcePVC` and `SouceSnapshotContent` to be clearer.

> If there's a way to dynamically display only one of SourcePVC and SourceSnapshotContent 
> depending on which one is not nil, that will be great. I don't know if it is possible to do that 
> using printerColumns.

Looking at https://book.kubebuilder.io/reference/markers/crd.html I don't see way to include conditionals when using `printcolumn`. 

> The source field will never be nil. Can we have a "VolumeSnapshotSource" field which is 
> either PVC or content name and have a "SourceType" field that indicates whether it is PVC 
> name or content name?

I'm assuming this would require a code change? I don't think this is possible currently from modifying `types.go`.

> It's probably more important to show the status content name since we expect them to be 
> the same in valid cases

Don't we do this presently in the PR?

```
// +kubebuilder:printcolumn:name="SnapshotContent",type=string,JSONPath=`.status.boundVolumeSnapshotContentName`,description="The name of the VolumeSnapshotContent to which this VolumeSnapshot is bound."
```